File: server/src/main/java/com/fluffy/oauth2/domain/OAuth2UserInfo.java
package com.fluffy.oauth2.domain;

import com.fluffy.auth.domain.Member;
import com.fluffy.auth.domain.OAuth2Provider;
import jakarta.annotation.Nullable;

public record OAuth2UserInfo(
        String socialId,
        String name,
        @Nullable String email,
        String avatarUrl
) {

    public Member toMember(OAuth2Provider provider) {
        return new Member(
                email,
                provider,
                socialId,
                name,
                avatarUrl
        );
    }
}

File: server/src/main/kotlin/com/fluffy/oauth2/domain/OAuth2UserInfo.kt
package com.fluffy.oauth2.domain

import com.fluffy.auth.domain.Member
import com.fluffy.auth.domain.OAuth2Provider


data class OAuth2UserInfo(
    val socialId: String,
    val name: String,
    val email: String?,
    val avatarUrl: String
) {

    fun toMember(provider: OAuth2Provider): Member {
        return Member(
            email = email,
            provider = provider,
            socialId = socialId,
            name = name,
            avatarUrl = avatarUrl
        )
    }
}
